src/GafferCycles/IECoreCyclesPreview/Renderer.cpp

```diff
@@ -3800,6 +3655,10 @@ class CyclesRenderer final : public IECoreScenePreview::Renderer
 		AttributesCachePtr m_attributesCache;
 
 		// Nodes created to update to Cycles
+		/// \todo I don't see why these need to be state on the Renderer.
```

Comment:
Yeah I'm curious to see how you approach this one, I put this here because it was a way to store newly-created nodes to hand off in a deferred single-threaded way under 1 lock and something done after any resets that needed to be made. It could live in the caches probably too as members, or we do scoped scene locks like the hydra delegate does and use the newer create_node calls that came after in later Cycles versions: https://github.com/blender/cycles/blob/main/src/scene/scene.h#L203

Reply:
Thanks for the info Alex. Initially I think I'll probably just squirrel these away inside the caches to help separate things out a bit, but longer term I definitely want to explore doing it on the fly with each thread taking the scene lock it needs it.